controller/appcontroller.go

```diff
@@ -665,6 +665,17 @@ func (ctrl *ApplicationController) processAppOperationQueueItem() (processNext b
 	}
 	app := origApp.DeepCopy()
 
+	if app.Operation != nil {
+		// If we get here, we are about process an operation but we cannot rely on informer since it might has stale data.
+		// So always retrieve the latest version to ensure it is not stale to avoid unnecessary syncing.
+		freshApp, err := ctrl.applicationClientset.ArgoprojV1alpha1().Applications(ctrl.namespace).Get(context.Background(), app.ObjectMeta.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Errorf("Failed to retrieve latest application state: %v", err)
+			return
+		}
+		app = freshApp
+	}
+
```

Comment:
minor difference, but I think it might be better to put this at the beginning of processRequestedAppOperation

Reply:
Actually nvmd. the other code would benefit from a fresh app

Reply:
Can you add a comment that this code can be deleted after https://github.com/argoproj/argo-cd/issues/6296 is fixed?

Reply:
sure